store/store.go

```diff
@@ -1190,6 +1200,29 @@ type buyError struct {
 	ErrorMessage string `json:"error_message"`
 }
 
+type storeError struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+func (s *storeError) Error() string {
+	return s.Message
+}
+
+type storeErrors struct {
+	Errors []*storeError `json:"error_list"`
+}
+
+func (s *storeErrors) Error() string {
+	message := ""
+
+	for _, e := range s.Errors {
+		message = message + e.Error() + "\n"
```

Comment:
This is a bit problematic. It will add a line feed when there's a single entry, which is not usual, and it will also format awkwardly when there are many.

To keep this simple and quick, let's please do something similar to what you did before:
```
if len(s.Errors) == 0 {
        return "internal error: empty store error used as an actual error"
}
return "store reported an error: " + s.Errors[0].Error()
```